resources/mediaplayer/src/test/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/wav/WavTrackCacheTest.java

```diff
@@ -0,0 +1,99 @@
+package org.mobicents.media.server.impl.resource.mediaplayer.audio.wav;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mobicents.media.server.impl.resource.mediaplayer.audio.CachedRemoteStreamProvider;
+import org.mobicents.media.server.impl.resource.mediaplayer.audio.DirectRemoteStreamProvider;
+import org.mobicents.media.server.spi.format.EncodingName;
+import org.mobicents.media.server.spi.format.Format;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import javax.sound.sampled.UnsupportedAudioFileException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
+ * Created by hamsterksu on 30.06.16.
+ */
+public class WavTrackCacheTest {
+
+    private long expectedDuration = 3854625000L;
+    private Format expectedFormat = new Format(new EncodingName("linear"));
+
+    URLStreamHandler handler;
+    URLConnection mockConnection;
+
+    @Before
+    public void setUp() throws IOException, UnsupportedAudioFileException {
+        mockConnection = mock(URLConnection.class);
+
+        //we need use answer to return new stream each time
+        when(mockConnection.getInputStream()).thenAnswer(new Answer<InputStream>() {
```

Comment:
we need to use this variant. otherwise answer will be cached and inputstream can't be read twice